Flaky test in TestAllocateWithPartitionPolicyNetworkTopology

### Description

Current there is a flaky test in `TestAllocateWithPartitionPolicyNetworkTopology`, case name is `rescheduling_scenario:_pending_tasks_with_different_resource_requirements` :

Here is the error log:
```
E0130 09:56:51.167556   41782 session.go:414] failed to find allocated hyperNode for subJob c1/pg1 by allocated nodes
W0130 09:56:52.099906   41782 allocate.go:678] The tier corresponding to highestTierName volcano.sh/hypercluster is not found, job <c1/pg1>
W0130 09:56:55.741931   41782 feature_gate.go:352] Setting GA feature gate DynamicResourceAllocation=true. It will be removed in a future release.
E0130 09:56:56.411259   41782 predicates.go:753] PreBind plugin VolumeBinding failed for pod c1/p1: binding volumes: failed to get node "n2": node "n2" not found
E0130 09:56:56.411551   41782 cache.go:1386] execute preBind for pod c1/p1 failed: binding volumes: failed to get node "n2": node "n2" not found, resync the task
E0130 09:56:56.411461   41782 predicates.go:753] PreBind plugin VolumeBinding failed for pod c1/p2: binding volumes: failed to get node "n2": node "n2" not found
E0130 09:56:56.411814   41782 cache.go:1386] execute preBind for pod c1/p2 failed: binding volumes: failed to get node "n2": node "n2" not found, resync the task
E0130 09:57:04.000641   41782 network_topology_aware.go:300] "build hyperNode gradient fail" err="getSearchRoot failed: there is no intersection between hyperNodeAvailable s0 and hyperNodeHighestAllowed s1" subJob="c1/pg1/task1-worker" hyperNode="s0" highestAllowedTier=1 allocatedHyperNode="s1"
E0130 09:57:04.959336   41782 network_topology_aware.go:300] "build hyperNode gradient fail" err="getSearchRoot failed: there is no intersection between hyperNodeAvailable s5 and hyperNodeHighestAllowed s1" subJob="c1/pg1/task1-worker" hyperNode="s5" highestAllowedTier=2 allocatedHyperNode="s3"
E0130 09:57:04.959559   41782 network_topology_aware.go:300] "build hyperNode gradient fail" err="getSearchRoot failed: there is no intersection between hyperNodeAvailable s6 and hyperNodeHighestAllowed s1" subJob="c1/pg1/task1-worker" hyperNode="s6" highestAllowedTier=2 allocatedHyperNode="s3"
E0130 09:57:05.440667   41782 network_topology_aware.go:300] "build hyperNode gradient fail" err="getSearchRoot failed: there is no intersection between hyperNodeAvailable s1 and hyperNodeHighestAllowed s0" subJob="c1/pg1/task1-worker" hyperNode="s1" highestAllowedTier=1 allocatedHyperNode="s0"
E0130 09:57:05.902003   41782 network_topology_aware.go:300] "build hyperNode gradient fail" err="getSearchRoot failed: there is no intersection between hyperNodeAvailable s6 and hyperNodeHighestAllowed s1" subJob="c1/pg1/task1-worker" hyperNode="s6" highestAllowedTier=2 allocatedHyperNode="s1"
E0130 09:57:05.908387   41782 network_topology_aware.go:300] "build hyperNode gradient fail" err="getSearchRoot failed: there is no intersection between hyperNodeAvailable s5 and hyperNodeHighestAllowed s1" subJob="c1/pg1/task1-worker" hyperNode="s5" highestAllowedTier=2 allocatedHyperNode="s1"
E0130 09:57:05.908917   41782 network_topology_aware.go:300] "build hyperNode gradient fail" err="getSearchRoot failed: there is no intersection between hyperNodeAvailable s2 and hyperNodeHighestAllowed s1" subJob="c1/pg1/task1-worker" hyperNode="s2" highestAllowedTier=2 allocatedHyperNode="s1"
E0130 09:57:09.251737   41782 network_topology_aware.go:300] "build hyperNode gradient fail" err="getSearchRoot failed: there is no intersection between hyperNodeAvailable s0 and hyperNodeHighestAllowed s1" subJob="c1/pg1/task1-worker" hyperNode="s0" highestAllowedTier=1 allocatedHyperNode="s1"
E0130 09:57:09.729919   41782 network_topology_aware.go:300] "build hyperNode gradient fail" err="getSearchRoot failed: there is no intersection between hyperNodeAvailable s0 and hyperNodeHighestAllowed s1" subJob="c1/pg1/task1-worker" hyperNode="s0" highestAllowedTier=1 allocatedHyperNode="s1"
E0130 09:57:11.172057   41782 network_topology_aware.go:300] "build hyperNode gradient fail" err="getSearchRoot failed: there is no intersection between hyperNodeAvailable s1 and hyperNodeHighestAllowed s0" subJob="c1/pg1/task1-worker" hyperNode="s1" highestAllowedTier=1 allocatedHyperNode="s0"
E0130 09:57:11.640006   41782 network_topology_aware.go:300] "build hyperNode gradient fail" err="getSearchRoot failed: there is no intersection between hyperNodeAvailable s1 and hyperNodeHighestAllowed s0" subJob="c1/pg1/task1-worker" hyperNode="s1" highestAllowedTier=1 allocatedHyperNode="s0"
E0130 09:57:12.129771   41782 network_topology_aware.go:300] "build hyperNode gradient fail" err="getSearchRoot failed: there is no intersection between hyperNodeAvailable s0 and hyperNodeHighestAllowed s1" subJob="c1/pg1/task2-worker" hyperNode="s0" highestAllowedTier=1 allocatedHyperNode="s1"
E0130 09:57:12.129978   41782 network_topology_aware.go:300] "build hyperNode gradient fail" err="getSearchRoot failed: there is no intersection between hyperNodeAvailable s1 and hyperNodeHighestAllowed s0" subJob="c1/pg1/task1-master" hyperNode="s1" highestAllowedTier=1 allocatedHyperNode="s0"
E0130 09:57:12.602131   41782 network_topology_aware.go:283] "build hyperNode gradient fail" err="getSearchRoot failed: get highest allowed hyperNode failed: allocated hyperNode <cluster-top-hypernode> tier is greater than highest allowed tier 1" job="c1/pg1" hyperNode="<cluster-top-hypernode>" highestAllowedTier=1 allocatedHyperNode="<cluster-top-hypernode>"
E0130 09:57:13.006901   41782 network_topology_aware.go:283] "build hyperNode gradient fail" err="getSearchRoot failed: get highest allowed hyperNode failed: allocated hyperNode <cluster-top-hypernode> tier is greater than highest allowed tier 1" job="c1/pg1" hyperNode="<cluster-top-hypernode>" highestAllowedTier=1 allocatedHyperNode="<cluster-top-hypernode>"
--- FAIL: TestAllocateWithPartitionPolicyNetworkTopology (17.11s)
    --- FAIL: TestAllocateWithPartitionPolicyNetworkTopology/rescheduling_scenario:_pending_tasks_with_different_resource_requirements (0.40s)
        allocate_test.go:5561: failed to get Bind request in case 33(rescheduling scenario: pending tasks with different resource requirements)
FAIL
FAIL	volcano.sh/volcano/pkg/scheduler/actions/allocate	38.028s
```



### Steps to reproduce the issue

1.
2.
3.


### Describe the results you received and expected

Solved the flaky test

### What version of Volcano are you using?

master

### Any other relevant information

_No response_

--- a/pkg/scheduler/framework/session.go
+++ b/pkg/scheduler/framework/session.go
@@ -26,7 +26,6 @@ import (
 	"context"
 	"fmt"
 	"maps"
-	"math"
 	"sort"
 	"sync"
 
@@ -415,7 +414,7 @@ func (ssn *Session) recoverAllocatedHyperNode(job *api.JobInfo, hyperNodeSet set
 				break
 			}
 		}
-		minimumHyperNode := getMinimumHyperNode(subJobAllocatedHyperNode, nodesByHyperNode)
+		minimumHyperNode := getLowestTierHyperNode(subJobAllocatedHyperNode, hyperNodes)
 
 		if subJob.AllocatedHyperNode != minimumHyperNode {
 			subJobUpdated = true
@@ -446,20 +445,26 @@ func (ssn *Session) recoverAllocatedHyperNode(job *api.JobInfo, hyperNodeSet set
 	}
 }
 
-func getMinimumHyperNode(hyperNodes sets.Set[string], nodesByHyperNode map[string]sets.Set[string]) string {
+func getLowestTierHyperNode(hyperNodes sets.Set[string], hyperNodeInfos api.HyperNodeInfoMap) string {
 	if hyperNodes == nil || hyperNodes.Len() == 0 {
 		return ""
 	}
 
-	minimumNodeSize := math.MaxInt
-	minimumHyperNode := ClusterTopHyperNode
+	var lowestTierHyperNode *api.HyperNodeInfo
 	for name := range hyperNodes {
-		if nodes, found := nodesByHyperNode[name]; found && nodes.Len() < minimumNodeSize {
-			minimumNodeSize = nodes.Len()
-			minimumHyperNode = name
+		if hn, found := hyperNodeInfos[name]; found {
+			if lowestTierHyperNode == nil || hn.Tier() < lowestTierHyperNode.Tier() {
+				lowestTierHyperNode = hn
+			}
 		}
 	}
-	return minimumHyperNode
+
+	if lowestTierHyperNode == nil {
+		klog.Warningf("No valid hypernodes found in hyperNodeInfos for the given set: %v", hyperNodes.UnsortedList())
+		return ""
+	}
+
+	return lowestTierHyperNode.Name
 }
 
 func (ssn *Session) parseHyperNodesTiers() {

